File: frontend/src/types/interview.ts
export interface Interview {
  id: number;
  candidate_id: number;
  job_id: number;
  interview_type: 'ai_chat' | 'ai_voice' | 'ai_both';
  status: 'scheduled' | 'in_progress' | 'completed' | 'cancelled' | 'expired';
  scheduled_at?: string;
  started_at?: string;
  completed_at?: string;
  duration_limit_min: number;
  language: string;
  total_questions: number;
  questions_asked: number;
  recording_url?: string;
  created_by?: number;
  created_at: string;
  updated_at: string;
}

export interface InterviewListResponse {
  items: Interview[];
  total: number;
  page: number;
  page_size: number;
}

export interface InterviewCreateRequest {
  candidate_id: number;
  job_id: number;
  interview_type?: string;
  scheduled_at?: string;
  duration_limit_min?: number;
  language?: string;
}

export interface TranscriptEntry {
  id: number;
  speaker: 'ai' | 'candidate' | 'system';
  message_type: 'text' | 'audio' | 'system';
  content: string;
  sequence_order: number;
  timestamp: string;
}

export interface ChatMessage {
  type: 'start' | 'message' | 'end' | 'ping' | 'reconnect';
  content: string;
}

export interface ChatResponse {
  type:
    | 'greeting' | 'response' | 'complete' | 'ended' | 'error' | 'candidate'
    | 'stream_start' | 'stream_chunk' | 'stream_end'
    | 'thinking' | 'pong' | 'reconnected';
  content: string;
  code?: string;
  is_complete?: boolean;
  question_number?: number;
  total_questions?: number;
  time_remaining_min?: number;
  conversation_history?: { role: string; content: string }[];
  current_question_index?: number;
  isStreaming?: boolean;
}

export type ConnectionState = 'connecting' | 'connected' | 'reconnecting' | 'disconnected';

export interface InterviewDetail extends Interview {
  questions: InterviewQuestion[];
  answers: InterviewAnswer[];
  transcripts: TranscriptEntry[];
}

export interface InterviewQuestion {
  id: number;
  question_text: string;
  question_type: string;
  difficulty: string;
  question_order: number;
  is_follow_up: boolean;
  asked_at?: string;
}

export interface InterviewAnswer {
  id: number;
  question_id: number;
  answer_text?: string;
  answer_mode: 'text' | 'voice';
  confidence_score?: number;
  answered_at: string;
}

File: backend/app/schemas/interview.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, Field

from app.models.interview import (
    InterviewType, InterviewStatus, AnswerMode, SpeakerType, MessageType,
)


class InterviewCreate(BaseModel):
    candidate_id: int
    job_id: int
    interview_type: InterviewType = InterviewType.AI_CHAT
    scheduled_at: Optional[datetime] = None
    duration_limit_min: int = Field(default=30, ge=5, le=120)
    language: str = Field(default="en", max_length=10)


class InterviewUpdate(BaseModel):
    status: Optional[InterviewStatus] = None
    scheduled_at: Optional[datetime] = None
    duration_limit_min: Optional[int] = None


class InterviewResponse(BaseModel):
    id: int
    candidate_id: int
    job_id: int
    interview_type: InterviewType
    status: InterviewStatus
    scheduled_at: Optional[datetime] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    duration_limit_min: int
    language: str
    total_questions: int
    questions_asked: int
    recording_url: Optional[str] = None
    created_by: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class InterviewListResponse(BaseModel):
    items: List[InterviewResponse]
    total: int
    page: int
    page_size: int


class ChatMessage(BaseModel):
    content: str = Field(min_length=1, max_length=5000)
    message_type: str = "text"


class ChatResponse(BaseModel):
    speaker: SpeakerType
    content: str
    message_type: MessageType
    timestamp: datetime


class TranscriptEntry(BaseModel):
    id: int
    speaker: SpeakerType
    message_type: MessageType
    content: str
    sequence_order: int
    timestamp: datetime

    model_config = {"from_attributes": True}


class InterviewQuestionResponse(BaseModel):
    id: int
    question_text: str
    question_type: str
    difficulty: str
    question_order: int
    is_follow_up: bool
    asked_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class InterviewAnswerResponse(BaseModel):
    id: int
    question_id: int
    answer_text: Optional[str] = None
    answer_mode: AnswerMode
    confidence_score: Optional[float] = None
    answered_at: datetime

    model_config = {"from_attributes": True}


class InterviewDetailResponse(InterviewResponse):
    questions: List[InterviewQuestionResponse] = []
    answers: List[InterviewAnswerResponse] = []
    transcripts: List[TranscriptEntry] = []
